### PR TITLE
Ear 1195 vertical tab component

### PR DESCRIPTION
**Pull request by farres1**
Creating reusable vertical tabs component for settings page
The component is passed a title, a columns value and a list of objects representing the items in the vertical tabs 

### How to review
Check that the vertical tabs component renders on the settings pages
Check that each of the tab items are displayed correctly
Check that changing to another tab causes the active tab to change (active tab is blue)
Check that both settings page URLs are correct
Check that the themes tab is not enabled on social surveys
Check that trying to access themes page on social surveys using the URL redirects to general settings page
Check that the previous /settings URL redirects to the general settings page with URL /settings/general

### CODE REVIEW

File: eq-author/src/App/settings/GeneralSettingsPage.js [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+    });
+  };
+
+  const [updateQuestionnaire] = useMutation(updateQuestionnaireMutation);
+  const [questionnaireTitle, setQuestionnaireTitle] = useState(title);
+  const [questionnaireShortTitle, setQuestionnaireShortTitle] = useState(
+    shortTitle
+  );
+
+  let match = useRouteMatch();
```
Comment (MattElcock) [inline eq-author/src/App/settings/GeneralSettingsPage.js:147]:
```suggestion
  const match = useRouteMatch();
```

File: eq-author/src/App/settings/ThemesPage.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+const StyledPanel = styled.div`
+  max-width: 97.5%;
+  padding: 1.3em;
+`;
+
+const ThemesPage = ({ questionnaire }) => {
+  const { type } = questionnaire;
+
+  let match = useRouteMatch();
```
Comment (MattElcock) [inline eq-author/src/App/settings/ThemesPage.js:48]:
```suggestion
  const match = useRouteMatch();
```

**Review by MattElcock — CHANGES_REQUESTED** [review]

File: eq-author/src/constants/theme.js [hunk 3]
```diff
@@ -48,6 +48,10 @@ export const disabledStyle = css`
   pointer-events: none;
 `;
 
+export const hoverStyle = css`
+  background: rgba(0, 0, 0, 0.2);
```
Comment (jasehumphr) [inline eq-author/src/constants/theme.js:52]: Why are we using RGB here?

**Review by jasehumphr — APPROVED** [review]
AKATG

**Review by MattElcock — APPROVED** [review]

**Pull request merged**